src/nbl/ui/CWindowWin32.cpp

```diff
@@ -485,6 +513,132 @@ namespace ui
 		return DefWindowProc(hWnd, message, wParam, lParam);
 	}
 
+	E_KEY_CODE CWindowWin32::getNablaKeyCodeFromNative(uint32_t nativeWindowsKeyCode)
+	{
+		nbl::ui::E_KEY_CODE nablaKeyCode = EKC_NONE;
+		switch (nativeWindowsKeyCode)
+		{
+		case VK_BACK:			nablaKeyCode = EKC_BACKSPACE; break;
+		case VK_TAB:			nablaKeyCode = EKC_TAB; break;
+		case VK_CLEAR:			nablaKeyCode = EKC_CLEAR; break;
+		case VK_RETURN:			nablaKeyCode = EKC_ENTER; break;
+		case VK_SHIFT:			[[fallthrough]];
+		case VK_LSHIFT:			[[fallthrough]];
```

Comment:
nabla needs separate enums for right shift vs left shift

Reply:
Yeah i did that temporarily bc for now i'm not sure how those events are sent, like why does windows even need VK_SHIFT if there are separate keys for l/r shifts, but if it sends `VK_SHIFT` in case there is only one shift on a keyboard, then what keyboard event should i push into the channel if nabla will only have the l/r shift event? I don't think making 3 nabla enums like `shift`, `left shift` and `right shift` is a good decision

Reply:
obvs same for alt/ctrl

Reply:
>  I don't think making 3 nabla enums like shift, left shift and right shift is a good decision

Maybe treat "left" as default, and only include some flags on the `IKeyboardInputChannel` whether the keyboard has two of shift/ctrl/alt